src/Raven.Client/Server/DatabaseTopology.cs

```diff
@@ -45,15 +117,36 @@ public DynamicJsonValue ToJson()
             {
                 [nameof(Members)] = new DynamicJsonArray(Members),
                 [nameof(Promotables)] = new DynamicJsonArray(Promotables),
-                [nameof(Watchers)] = new DynamicJsonArray(Watchers)
+                [nameof(Watchers)] = new DynamicJsonArray(Watchers.Select(w => w.ToJson())),
+                [nameof(CustomConnectionBlocker)] = DynamicJsonValue.Convert(CustomConnectionBlocker),
+                [nameof(NameToUrlMap)] = DynamicJsonValue.Convert(NameToUrlMap)
             };
         }
 
         public void RemoveFromTopology(string delDbFromNode)
         {
             Members.Remove(delDbFromNode);
             Promotables.Remove(delDbFromNode);
-            Watchers.Remove(delDbFromNode);
+            CustomConnectionBlocker.Remove(delDbFromNode);
+        }
+
+        public bool IsItMyTask(IDatabaseTask task, string nodeTag)
+        {
+            var myPosition = Members.FindIndex(s => s == nodeTag);
+            return JumpConsistentHash((ulong)task.GetHashCode(), Members.Count) == myPosition;
```

Comment:
All the logic with the suspects I thought to implement in an other ticket, since this issue is already touches a lot of files and have many changes.